### PR TITLE
Fix and improve docs

### PR DESCRIPTION
**Pull request by xokker**
1. More detailed release instructions.
2. Fixed path to `env.secrets`.
3. More detailed dockerize instructions.
4. Expanded note on versioning and dedicated a section number 0 to it.

### CODE REVIEW

File: Documentation/Publishing.md [hunk 1]
```diff
@@ -25,4 +25,15 @@ To sign and publish the artifacts run the following Gradle command:
...
 ## 4. Release
-You can then release the deployment to the Central Repository following the steps on the Sonatype website  [here](https://central.sonatype.org/publish/release/). We are on the legacy host so make sure to use https://oss.sonatype.org/. Only attempt to Close the staging repository once the upload has finished.
\ No newline at end of file
+You can then release the deployment to the Central Repository following the steps on the Sonatype website  [here](https://central.sonatype.org/publish/release/). We are on the legacy host so make sure to use https://oss.sonatype.org/. Only attempt to Close the staging repository once the upload has finished.
+ 
+## 5. Create github release and tag
+
+1. Go to [/releases](https://github.com/google/data-transfer-project/releases);
+2. Click 'Draft a new release';
+3. fill in the title and create a tag (the tag must have a `v` prefix, the title mustn't);
```
Comment (jzacsh) [inline Documentation/Publishing.md:34]: I'm imagining we'd prefer semver releases - should we just say that explicitly?

Comment (jzacsh) [reply Documentation/Publishing.md:34]: Or maybe call out that things are still in flux[^1] and say something _vaguely_ about semver; eg: _"step 0: start a discussion in maintainers chat about setting a new release (until the process is better documented); step 3: pick a new release version name by referring to our previous release, according to semver.org values (major, minor, patch)"_

[^1]: I think you said you were going to reachout in maintainers room with a doc or something? SGTM!

**Review by jzacsh — CHANGES_REQUESTED** [review]
LGTM - thanks for cleaning things up. just a clarifying question below about releases.

File: Documentation/Publishing.md (hunk 1 above)
Comment (xokker) [reply Documentation/Publishing.md]: Added notes on versioning as a separate section. What do you think about the updated version?

**Review by wmorland — APPROVED** [review]
LGTM

File: Documentation/Publishing.md (hunk 1 above)
Comment (wmorland) [reply Documentation/Publishing.md]: Discussed in the call on Monday

**Review by jzacsh — APPROVED** [review]

**Pull request merged**